### PR TITLE
Warn organization owner if signup is enabled but not valid

### LINKED ISSUE #9778: Warn organization owner if signup is enabled but not valid
If organization signup is enabled (authentication enabled and whitelisting emails) but not valid (because seat or quota limit has been reached) organization settings page should display an error.

Some design magic is needed (@josecruz ?).

cc @yasharora73 @gfiorav @javitonino @xavijam

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes #9778.

It involves flash messages (user general error messages) improvements, as this validation is a warning (first time needed) and it might display both a success and a warning.

Some screenshots:

captura de pantalla 2017-01-25 a las 18 07 15
captura de pantalla 2017-01-25 a las 18 07 27
captura de pantalla 2017-01-25 a las 18 07 50
captura de pantalla 2017-01-25 a las 18 08 04

Acceptance minimum:
- [ ] Organization with authentication, whitelisted domains and available quota and seats should display no message.
- [ ] Without remaining seats it should display message.
- [ ] Without remaining quota it should display both messages.
- [ ] Without authentication it should hide the messages.
- [ ] Successful organization signup with valid configuration.
- [ ] Organization signup error without remaining quota.
- [ ] Organization signup error without remaining seats.
- [ ] Organization groups page flash message.

Acceptance regression:
- [ ] Find other flash messages (banner notifications) through the application and ensure they display correctly. For example, when an update to user parameters fails (e.g: invalid email)

### CODE REVIEW

File: spec/requests/admin/organizations_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        @organization.save
+
+        get organization_auth_url(user_domain: @org_user_owner.username)
+
+        response.status.should eq 200
+        response.body.should include("Your organization has run out of seats")
+
+        @organization.seats = old_seats
+        @organization.save
+      end
```
Comment (javitonino) [inline spec/requests/admin/organizations_controller_spec.rb]: What happens if both messages should be displayed at the same time. I see it is not on the tests nor on the screenshots. According to the code, it should concatenate both messages. Is the end result reasonable, i.e: does not break the styles with a long line?

File: app/models/organization.rb [hunk 2]
```diff
@@ -88,19 +88,27 @@ def validate_new_user(user, errors)
...
+    if !validate_disk_quota(user.quota_in_bytes.to_i)
       errors.add(:quota_in_bytes, "not enough disk quota")
     end
   end
 
+  def validate_disk_quota(quota = default_quota_in_bytes)
+    unassigned_quota >= quota
+  end
+
+  def validate_builder_seats(users = [])
```
Comment (javitonino) [inline app/models/organization.rb]: These couple of methods return a boolean, so maybe rename to `valid_disk_quota?` or even `available_disk_quota?`? When I see validate, my first impression is that this was a validator, and that it wrote to `errors` if invalid.

File: lib/build/tasks/watch.js [hunk 3]
```diff
@@ -88,6 +88,7 @@ exports.task = function () {
     css: {
       files: [
         'app/assets/stylesheets/editor-3/**/*.css.scss',
+        // 'app/assets/stylesheets/common/**/*.css.scss',
```
Comment (juanignaciosl) [inline lib/build/tasks/watch.js]: Yes, I took @matallo suggestion about keeping it there, commented. I'll add a `// Uncomment next line to watch Dashboard and Editor SCSS files` to make it clear (open to improvement suggestions).

File: spec/requests/admin/organizations_controller_spec.rb (hunk 1 above)
Comment (juanignaciosl) [inline spec/requests/admin/organizations_controller_spec.rb]: It Works™:

captura de pantalla 2017-01-26 a las 15 16 51

File: app/models/organization.rb (hunk 2 above)
Comment (juanignaciosl) [inline app/models/organization.rb]: Good point 👍

**Comment by juanignaciosl** [comment]
Moving to acceptance on 🍏  then!

**Comment by juanignaciosl** [comment]
Only a minor change. @ivanmalagon , could you please check whatever you already did (I think that all but one) so the next acceptance focuses only in the remaining ones?

**Pull request merged**